dss/storage/index_document.py

```diff
@@ -191,6 +191,47 @@ def get_shape_descriptor(self) -> typing.Optional[str]:
         else:
             return None  # No files with schema identifiers were found
 
+    def get_indexed_versions(self) -> typing.MutableMapping[str, str]:
+        """
+        Returns a dictionary mapping the name of each index containing this document to the
+        version of this document in that index. Note that `version` denotes document version, not
+        bundle version.
+        """
+        page_size = 64
+        es_client = ElasticsearchClient.get(self.logger)
+        alias_name = Config.get_es_alias_name(ESIndexType.docs, self.replica)
+        response = es_client.search(index=alias_name, body={
+            '_source': False,
+            'stored_fields': [],
+            'version': True,
+            'from': 0,
+            'size': page_size,
+            'query': {
+                'terms': {
+                    '_id': [self.bundle_fqid]
+                }
+            }
+        })
+        hits = response['hits']
+        assert hits['total'] <= page_size, 'Document is in too many indices'
```

Comment:
Could you help me understand how this works a little more? I would have expected that the above query would return just one document per versioned index

Reply:
I expect one hit per presence of a document with the given id in any index. I expect there to be a handful of indices and things would have to go seriously wrong for an id to be in all of them, so 64 seemed like a safe upper bound. I just didn't want to write code for paging and test it. Maybe the assertion message isn't clear enough.

Reply:
I see. Better to fail fast in such an event, I suppose.